files/Aboot/boot0.j2

```diff
@@ -317,106 +445,155 @@ platform_specific() {
-        # do not have the $cmdline_base file. we assume that we are on Sonic or EOS.
-        cat /proc/cmdline | sed -E 's/^(.*) rw .*$/\1/' | tr ' ' '\n' > $cmdline_base
-    fi
+    # fs configuration
+    cmdline_add rw
 
-    cp $cmdline_base /tmp/append
+    # disable deterministic interface naming
+    cmdline_add net.ifnames=0
 
-    platform_specific
-    echo "rw loop=$image_name/fs.squashfs loopfstype=squashfs apparmor=1 security=apparmor quiet" >> /tmp/append
+    # verbosity
+    cmdline_add quiet
+    # Start showing systemd information from the first failing unit if any.
+    # systemd.show_status=false or quiet can be used to silence systemd entierly
+    cmdline_add systemd.show_status=auto
 
     # Pass the MAC address to the new kernel as a command line parameter. This makes it
     # possible to restore the MAC address in the new kernel without requiring driver modifications.
     if [ -f /sys/class/net/ma1/address ]; then
-        echo "hwaddr_ma1=$(cat /sys/class/net/ma1/address)" >> /tmp/append
+        cmdline_add "hwaddr_ma1=$(cat /sys/class/net/ma1/address)"
     elif [ -f /sys/class/net/eth0/address ]; then
-        echo "hwaddr_ma1=$(cat /sys/class/net/eth0/address)" >> /tmp/append
+        cmdline_add "hwaddr_ma1=$(cat /sys/class/net/eth0/address)"
     else
         err "Management port not found."
     fi
 
+    # Obtain root partition uuid
+    local rootdev="$(mount | grep -E "$flash_re" | cut -f1 -d' ')"
+    local rootfstype="$(mount | grep -E "$flash_re" | cut -f5 -d' ')"
+    local rootuuid="$(get_uuid_for $rootdev)"
+    if [ -z "$rootuuid" ] || [ "$rootfstype" = "vfat" ] ; then
+        cmdline_add "root=$rootdev"
+    else
+        cmdline_add "root=UUID=$rootuuid"
+    fi
+}
+
+write_default_cmdline() {
+    local delimiter="cmdline-aboot-end"
+
+    cmdline_clear
+
+    if $in_aboot; then
+        # generate the default kernel parameters for the platform
+        cat /etc/cmdline | sed "/^\(${bootconfigvars// /\|}\|crashkernel\|loglevel\|ignore_loglevel\)\(\$\|=\)/d;/^\$/d" | cmdline_append
+    elif grep -q "$delimiter" /proc/cmdline; then
+        # we are on a recent sonic image using delimiter. extracting the part of the
+        # cmdline coming from aboot is trivial.
+        cat /proc/cmdline | sed -E "s/^(.*) $delimiter .*$/\1/" | tr ' ' '\n' | cmdline_append
+    else
+        # we are either on SONiC or EOS and the commandline doesn't have a delimiter
+        # for the Aboot part. Take an educated guess at a right delimiter.
+        # Subject to breakage if EOS or SONiC cmdline change.
+        cat /proc/cmdline | sed -E 's/^(.*) rw .*$/\1/' | tr ' ' '\n' | cmdline_append
+    fi
+
+    cmdline_add "$delimiter"
+}
+
+write_common_configs() {
+    write_default_cmdline
+    write_platform_specific_cmdline
+    write_image_specific_cmdline
+    write_machine_config
+}
+
+write_secureboot_configs() {
+    write_common_configs
+    cmdline_add "loop=${installer_image_path/mnt\/flash\//}"
```

Comment:
Kernel panic, not quite sure if it is caused by /mnt/flash specified.

116.70: Next reboot will use flash:image-master.0-0f2b34b8/sonic.swi
116.95: Kexecing...
[  117.157009] kexec_core: Starting new kernel
zvln6t3tALERT! /host/mnt/flash/image-master.0-0f2b34b8/sonic.swi does not exist.  Dropping to a shell!
Rebooting automatically due to panic= boot argument
[    2.349976] Kernel panic - not syncing: Attempted to kill init! exitcode=0x00000000
[    2.349976] 
[    2.459633] CPU: 3 PID: 1 Comm: init Tainted: G            E     4.19.0-6-amd64 #1 Debian 4.19.67-2+deb10u2
[    2.576585] Hardware name: Arista Woodpecker/Woodpecker, BIOS Aboot-norcal9-9.0.1-11845100 04/07/2019
[    2.687179] Call Trace:

Reply:
I believe the intention of this statement is trying to remove `/mnt/flash` inside `$installer_image_path`. If @xumia agree, could you help test this fix in an image?

---
In reply to: [440555839](https://github.com/Azure/sonic-buildimage/pull/4741#discussion_r440555839) [](ancestors = 440555839)

Reply:
Can we change a way not relative to shell env? none of them works correctly.
Aboot# installer_image_path=/mnt/flash/image-master.0-0f2b34b8/sonic.swi
Aboot# echo "loop=${installer_image_path/mnt\/flash\//}"
loop=/mnt/flash/image-master.0-0f2b34b8/sonic.swi
Aboot# echo "loop=${installer_image_path#mnt/flash}"
loop=/mnt/flash/image-master.0-0f2b34b8/sonic.swi
Aboot# echo "loop=${installer_image_path#mnt/flash#}"
loop=/mnt/flash/image-master.0-0f2b34b8/sonic.swi
Aboot#

Reply:
I will investigate this issue.

Reply:
I only tested my changes on Aboot 9.0.3 and 9.0.4 where it's working.
I checked and was able to reproduce on 9.0.1.
We upgraded busybox in Aboot and the older version doesn't seem to support this shell syntax.
I'll use sed instead.